js/stores/appStore.js

```diff
@@ -80,8 +80,27 @@ class AppStore extends EventEmitter {
 
   emitChanges () {
     if (this.lastEmittedState && this.lastEmittedState !== appState) {
-      const d = diff(this.lastEmittedState, appState)
-      if (!d.isEmpty()) {
+      let d
+      try {
+        d = diff(this.lastEmittedState, appState)
+      } catch (e) {
+        console.error('Error getting a diff from latest state.')
+        // one possible reason immutablediff can throw an error
+        // is due to null keys, so let's log any that we find
+        const nullKeyPaths = findNullKeyPaths(appState)
+        const error = (typeof e === 'object')
```

Comment:
I think we should be more specific about the type of error before we do a null key check. We don't want to assume that all exceptions are this specific issue.

Reply:
The trouble is that immutablediff does not throw an error object, just a (dynamic) string. This is the only case I've seen immutablediff fail so far, and I did not want to go down the route of either:
- parsing the string, since it could change in a later immutablediff version
- assuming string vs Error object would mean a null-key error, since that could also change

Again, since the app will be unusable in this state, I wanted to err on the side of caution with regards to making sure we get the right error output and reported, so that we can solve the issue. But happy to take advice on this.